### PR TITLE
Histolab BinaryMask module

### LINKED ISSUE #158: Add option to change extraction masks
Currently, Histolab extracts tiles from the biggest tissue box of the slide (for backward compatibility with previous work).

I'd like to have the possibility to change the area to extract from, for example from the segmented tissue area.

Another cool thing to add could be the possibility for the users to supply their own binary mask for the extraction.

To achieve this, there are two viable options:
1) add a parameter to the extract method of the tilers, like 'mode' or something like this, to control the extraction area. This would be easy and compatible with the previous API, but users must use one of the modes we are providing and cannot implement a custom one
2) provide a different API to provide a mask, either one of the Histolab masks or a custom extraction mask

### PR DESCRIPTION
**Pull request by ernestoarbitrio**
- Close #158
- Deprecate `.biggest_tissue_box_mas` in `Slide`
- Add `TissueMask` object as new binary mask calculation

### CODE REVIEW

File: .isort.cfg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[tool.isort]
+profile = "black"
```
[the remark this answers is not in the document]
Comment (ernestoarbitrio) [reply .isort.cfg:2]: ????

File: src/histolab/masks.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            raise ValueError(
+                f"n should be smaller than the number of regions [{len(regions)}], "
+                f"got {n}"
+            )
+
+        sorted_regions = sorted(regions, key=lambda r: r.area, reverse=True)
+        return sorted_regions[:n]
+
+
+class TissueMask(BinaryMask):
```
[the remark this answers is not in the document]
Comment (ernestoarbitrio) [reply src/histolab/masks.py:111]: **nope, i'll add a custom sphinx rst file for this**

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -1,10 +1,26 @@
...
-    - id: flake8
\ No newline at end of file
+      - id: flake8
+        additional_dependencies: [ "flake8-eradicate==0.4.0" ]
+
+  - repo: https://github.com/pycqa/isort
+    rev: 5.6.4
+    hooks:
+      - id: isort
+        args: ["--profile", "black"]
```
[the remark this answers is not in the document]
Comment (ernestoarbitrio) [reply .pre-commit-config.yaml:18]: i remove the isort.cfg and keep this one ... i do prefer this way

File: src/histolab/tiler.py [hunk 4]
```diff
@@ -271,6 +280,7 @@ def extract(
             Slide from which to extract the tiles
         extraction_mask : BinaryMask
             BinaryMask object defining how to compute a binary mask from a Slide.
+            Default `BiggestTissueBoxMask`.
```
[the remark this answers is not in the document]
Comment (ernestoarbitrio) [reply src/histolab/tiler.py:283]: done

**Review by alessiamarcolini — APPROVED** [review]
LGTM

**Pull request merged**